NOT A NIXPKGS BUG nor the cause of any failure stdenv: `pop_var_context: head of shell_variables not a function context`

Not a nixpkgs bug nor the cause of any failure
https://mail.gnu.org/archive/html/bug-bash/2022-10/msg00075.html


```
/nix/store/559pz0w6zlvw8yyxah9s10fhaz400vaj-stdenv-darwin/setup: line 138: pop_var_context: head of shell_variables not a function context
```

Hi @ciferkey, this issue's topic is unrelated to the error you're speaking of. See https://github.com/NixOS/nixpkgs/pull/207083#issuecomment-1370381708 -- the line in this issue can be ignored.

The actual issue is *above* that line in the pahelper build you're talking about:

```
patoolib/__init__.py:560: PatoolError
=========================== short test summary info ============================
FAILED tests/test_mime.py::TestMime::test_mime_file_bzip - AssertionError: 'application/x-bzip2' != 'application/x-tar'
FAILED tests/test_mime.py::TestMime::test_mime_file_compress - AssertionError: 'application/x-compress' != 'application/x-tar'
FAILED tests/test_mime.py::TestMime::test_mime_file_lzip - AssertionError: 'application/x-lzip' != 'application/x-tar'
FAILED tests/test_mime.py::TestMime::test_mime_file_xzip - AssertionError: 'application/x-xz' != 'application/x-tar'
FAILED tests/archives/test_pytarfile.py::TestPytarfile::test_py_tarfile_bz2_file - patoolib.util.PatoolError: module 'patoolib.programs.py_tarfile' has no att...
FAILED tests/archives/test_tar.py::TestTar::test_tar_bz2_file - patoolib.util.PatoolError: module 'patoolib.programs.tar' has no attribute ...
FAILED tests/archives/test_tar.py::TestTar::test_tar_lzip_file - patoolib.util.PatoolError: module 'patoolib.programs.tar' has no attribute ...
FAILED tests/archives/test_tar.py::TestTar::test_tar_xz_file - patoolib.util.PatoolError: module 'patoolib.programs.tar' has no attribute ...
=========== 8 failed, 64 passed, 98 skipped, 4 deselected in 19.22s ============
/nix/store/b09v23lirgvci3wzszh22mbkdfj0h0yq-stdenv-linux/setup: line 1582: pop_var_context: head of shell_variables not a function context
```

--- a/pkgs/shells/bash/5.nix
+++ b/pkgs/shells/bash/5.nix
@@ -61,6 +61,8 @@ stdenv.mkDerivation rec {
     #   https://savannah.gnu.org/patch/index.php?10373
     # Had to fetch manually to workaround -p0 default.
     ./parallel.patch
+    # Fix `pop_var_context: head of shell_variables not a function context`.
+    ./fix-pop-var-context-error.patch
   ];
 
   configureFlags = [

--- a/pkgs/shells/bash/5.nix
+++ b/pkgs/shells/bash/5.nix
@@ -57,11 +57,6 @@ stdenv.mkDerivation rec {
 
   patches = upstreamPatches ++ [
     ./pgrp-pipe-5.patch
-    (fetchurl {
-      name = "fix-static.patch";
-      url = "https://cgit.freebsd.org/ports/plain/shells/bash/files/patch-configure?id=3e147a1f594751a68fea00a28090d0792bee0b51";
-      sha256 = "XHFMQ6eXTReNoywdETyrfQEv1rKF8+XFbQZP4YoVKFk=";
-    })
     # Apply parallel build fix pending upstream inclusion:
     #   https://savannah.gnu.org/patch/index.php?10373
     # Had to fetch manually to workaround -p0 default.

--- a/pkgs/shells/bash/bash-5.2-patches.nix
+++ b/pkgs/shells/bash/bash-5.2-patches.nix
@@ -27,4 +27,10 @@ patch: [
 (patch "024" "1hq23djqbr7s9y2324jq9mxr5bwdkmgizn3zgpchbsqp054k85cp")
 (patch "025" "0x9hc4silzl4d3zw4p43i5dm7w86k50j47f87lracwfgwy3z8f2i")
 (patch "026" "1b1fhm1dsi67r8ip17s0xvx2qq31fsxc1g9n3r931dd0k9a1zvln")
+(patch "027" "0fdbhvs9dkf4knncifh98a76q4gylhyvfrffq5p9q3ag5q58jap1")
+(patch "028" "1hdacd6sssjshmry1sscdnxxfb2r51bvdyghlfjaqgc9l85phhk0")
+(patch "029" "11wrlb20w6v89b96krg0gwxipwhvrda6rq1y9f972m32gsrsqp0j")
+(patch "030" "13v9fqgim082dmvkslsr0hs793yzhsij2s91mjswsfhj1qip7zy3")
+(patch "031" "15d7rddj6spwc1fy997lxx6zvzq0zbxgf2h20mhi4wgp5nzbglf2")
+(patch "032" "05ia6yf32hjprmyyxqawhgckxs3684ikfx8xg08zfgx9xkd7g73v")
 ]

--- a/pkgs/shells/bash/fix-pop-var-context-error.patch
+++ b/pkgs/shells/bash/fix-pop-var-context-error.patch
@@ -1,0 +1,17 @@
+Excerpted from <https://git.savannah.gnu.org/cgit/bash.git/commit/?h=devel&id=055a4552c901e43e6a6007f5bc664a33698dbf85>.
+
+Original author: Chet Ramey <chet.ramey@case.edu>
+
+--- variables.c
++++ variables.c
+@@ -5413,7 +5413,9 @@ pop_var_context ()
+   vcxt = shell_variables;
+   if (vc_isfuncenv (vcxt) == 0)
+     {
+-      internal_error (_("pop_var_context: head of shell_variables not a function context"));
++      /* If we haven't flushed all of the local contexts already, flag an error */
++      if (shell_variables != global_variables || variable_context > 0)
++	internal_error (_("pop_var_context: head of shell_variables not a function context"));
+       return;
+     }
+ 

